src/Hmvc/tests/CoreTest.php

```diff
@@ -7,191 +7,232 @@
 use PHPUnit\Framework\TestCase;
 use Spiral\Core\Container;
 use Spiral\Core\Exception\ControllerException;
+use Spiral\Core\Scope;
+use Spiral\Framework\ScopeName;
 use Spiral\Tests\Core\Fixtures\CleanController;
 use Spiral\Tests\Core\Fixtures\DummyController;
 use Spiral\Tests\Core\Fixtures\SampleCore;
 
-class CoreTest extends TestCase
+final class CoreTest extends TestCase
 {
+    private Container $root;
+
+    protected function setUp(): void
+    {
+        $this->root = new Container();
+    }
+
     public function testCallAction(): void
     {
-        $core = new SampleCore(new Container());
-        $this->assertSame('Hello, Antony.', $core->callAction(
-            DummyController::class,
-            'index',
-            ['name' => 'Antony']
-        ));
+        $core = new SampleCore($this->root);
+
+        $this->root->runScope(new Scope(ScopeName::Http), function () use ($core): void {
```

Comment:
we have to simplify this for users, it's ok in framework, but people won't be creating HTTP scope by themselves.

Reply:
also, the test can be in TEST scope overall, most likely "HTTP" scope won't be doing much, HTTP.request will.